var/spack/repos/builtin/packages/samrai/package.py

```diff
@@ -104,4 +103,16 @@ def configure_args(self):
                 '--enable-opt',
                 '--disable-debug'])
 
+        if self.version >= Version('3.12.0'):
+            # only version 3.12.0 and above, samrai does not use
+            # boost, but needs c++11. Without c++11 flags, samrai
+            # cannot build with either gcc or intel compilers.
+            if 'CXXFLAGS' in env and env['CXXFLAGS']:
+                env['CXXFLAGS'] += ' ' + self.compiler.cxx11_flag
+            else:
+                env['CXXFLAGS'] = self.compiler.cxx11_flag
+        else:
+            # boost 1.64.0 or earlier works with samrai 2.4.4~3.11.5
+            options.extend(['--with-boost=%s' % self.spec['boost'].prefix])
```

Comment:
I would use `options.append()` instead of `options.extend([])`. No need to create an extra list.

Reply:
Also, this is already being added above.

Reply:
The above one was intended to be removed.

        if self.spec.satisfies('@0:3.11.99'):
            options.extend([
                '--with-boost=%s' % self.spec['boost'].prefix
            ])